rfcs/testdriver_cross_origin.md

```diff
@@ -0,0 +1,70 @@
+# RFC 63
+
+## Summary
+Support running testdriver commands in browsing contexts other than
+the test browsing context, including cross origin contexts where it's
+possible to get a handle to the browsing context containing the test
+(i.e. excluding the rel=noopener case).
+
+## Details
+
+For contexts in which it's possible to get a handle to the context
+containing testharness.js ("the test context"), we can support
+testdriver interactions by using postMessage to the top-level window
+with a message containing the command to run and the context to run it
+in. That approach works with the WebDriver-based implementation in
+wptrunner where WebDriver can only address a single frame at a time,
+and a WebDriver-driven event loop in the test context is used to
+invoke testdriver actions and also to report the test results.
+
+The specific proposed changes are:
+
+* Allow testdriver.js to be loaded in contexts other than the test
+  context.
+
+* In such contexts require use of testdriver to first call the
```

Comment:
I wonder if we should not make this (and `message_test`) part of the public contract of testdriver.js. Instead, we would specify that `test_driver_internal` should work in non-test browsing contexts and leave it to the vendors how they would support that?

Reply:
You're very correct, of course. I started with "I want to eliminate `message_test` from the public API" and then somehow let my brain increase the scope to `set_test_context` despite the fact that part has to be called by test authors -_-.

Reply:
I think `message_test` is useful because even with a perfect system you need some way to tell the test window that the testdriver action completed.

Reply:
Ack, sgtm.